### PR TITLE
Add a method to remove host from account stats database

### PR DESCRIPTION
**Pull request by justinlin-linkedin**
This PR adds a new method to remove a specific host from account stats database. This would be useful when a host is removed clustermap.

### CODE REVIEW

File: ambry-mysql/src/main/java/com/github/ambry/accountstats/AccountReportsDao.java [hunk 1]
```diff
@@ -157,6 +159,29 @@ void queryStorageUsageForHost(String clusterName, String hostname, ContainerStor
...
+        long startTimeMs = System.currentTimeMillis();
+        deleteStatement.setString(1, clusterName);
+        deleteStatement.setString(2, hostname);
+        deleteStatement.executeUpdate();
+        metrics.deleteTimeMs.update(System.currentTimeMillis() - startTimeMs);
+        metrics.deleteFailureCount.inc();
+      }
+    } catch (SQLException e) {
+      metrics.deleteFailureCount.inc();
+      logger.error("Failed to execute DELETE on {}, with parameter {}", ACCOUNT_REPORTS_TABLE, hostname, e);
```
Comment (ankagrawal) [inline ambry-mysql/src/main/java/com/github/ambry/accountstats/AccountReportsDao.java:180]: This log statement won't print the exception "e".

Maybe change this to 
```logger.error("Failed to execute DELETE on {}, with parameters {}, {} due to exception: {}", ACCOUNT_REPORTS_TABLE, hostname, clusterName, e);```

**Review by ankagrawal — COMMENTED** [review]
Looks good to me overall.
Please fix the logging format.

File: ambry-mysql/src/main/java/com/github/ambry/accountstats/AccountReportsDao.java (hunk 1 above)
Comment (ankagrawal) [reply ambry-mysql/src/main/java/com/github/ambry/accountstats/AccountReportsDao.java:180]: nit: while we are at it, maybe change the `hostname` param to `hostName` (or `clusterName` to `clustername`).

Comment (justinlin-linkedin) [reply ambry-mysql/src/main/java/com/github/ambry/accountstats/AccountReportsDao.java:180]: 1. logger would check if the last argument is a throwable, if it's, logger would trim it out of the argument list and still print it.
2. sure, update clusterName to clustername.

File: ambry-mysql/src/main/java/com/github/ambry/accountstats/AccountReportsDao.java [hunk 2]
```diff
@@ -151,92 +153,115 @@ void queryStorageUsageForHost(String clusterName, String hostname, ContainerStor
...
+   */
+  void deleteStorageUsageForHost(String clustername, String hostname) throws SQLException {
+    try (Connection connection = dataSource.getConnection()) {
+      try (PreparedStatement deleteStatement = connection.prepareStatement(deleteHostSql)) {
+        long startTimeMs = System.currentTimeMillis();
+        deleteStatement.setString(1, clustername);
+        deleteStatement.setString(2, hostname);
+        deleteStatement.executeUpdate();
+        metrics.deleteTimeMs.update(System.currentTimeMillis() - startTimeMs);
+        metrics.deleteFailureCount.inc();
```
Comment (Arun-LinkedIn) [inline ambry-mysql/src/main/java/com/github/ambry/accountstats/AccountReportsDao.java:176]: Typo for `deleteSuccessCount`.

File: ambry-api/src/main/java/com/github/ambry/accountstats/AccountStatsStore.java [hunk 3]
```diff
@@ -31,6 +32,14 @@ public interface AccountStatsStore {
    */
   void storeHostAccountStorageStats(HostAccountStorageStatsWrapper statsWrapper) throws Exception;
 
+  /**
+   * Delete {@link HostAccountStorageStats} for the given host.
+   * @param hostname The hostname
+   * @param port The port
+   * @throws Exception
+   */
+  void deleteHostAccountStorageStatsForHost(String hostname, int port) throws Exception;
```
Comment (Arun-LinkedIn) [inline ambry-api/src/main/java/com/github/ambry/accountstats/AccountStatsStore.java:41]: Minor. Can method name be `deleteHostAccountStorageStats`?

File: ambry-mysql/src/main/java/com/github/ambry/accountstats/AccountReportsDao.java (hunk 2 above)
Comment (justinlin-linkedin) [reply ambry-mysql/src/main/java/com/github/ambry/accountstats/AccountReportsDao.java]: updated

File: ambry-api/src/main/java/com/github/ambry/accountstats/AccountStatsStore.java (hunk 3 above)
Comment (justinlin-linkedin) [reply ambry-api/src/main/java/com/github/ambry/accountstats/AccountStatsStore.java:41]: I think the method name is fine. HostAccountStorageStats is a term we use to represent the account stats reported by host. We have other method that delete this for cluster, for container, account etc...

Comment (Arun-LinkedIn) [reply ambry-api/src/main/java/com/github/ambry/accountstats/AccountStatsStore.java:41]: Sure.

**Review by Arun-LinkedIn — APPROVED** [review]

**Review by ankagrawal — APPROVED** [review]

**Pull request merged**